main.cpp

```diff
@@ -38,18 +41,53 @@ int main(int argc, char *const argv[]) {
     }
 
     const IR::P4Program *program = nullptr;
-
+    // std::cout << "\nChecking " << options.file;
+    cstring command = "python3 /home/roborobo/projects/gauntlet_paper/gauntlet/"
+                      "validate_p4_translation.py -i ";
+    command += realpath(options.file, NULL);
+    // command += " > /dev/null";
+    // command += options.file;
+    // std::wcout << command;
+    int required_exit_code = system(command.c_str());
+    printf("%s\n", command);
+    std::cout << "Got code :" << required_exit_code << " for the main file";
     program = P4::parseP4File(options);
 
     if (program != nullptr && ::errorCount() == 0) {
         // P4::ToP4 *before = new P4::ToP4(&std::cout, false);
         // program->apply(*before);
 
-        P4PRUNER::Pruner *pruner = new P4PRUNER::Pruner();
-        program = program->apply(*pruner);
+        for (int j = 0; j < 10; j++) {
+            P4PRUNER::Pruner *pruner = new P4PRUNER::Pruner();
+            for (int i = 0; i < 100; i++) {
+                auto temp = program;
+                temp = temp->apply(*pruner);
+                auto temp_f = new std::ofstream("temp.p4");
```

Comment:
Also this tmp file should be in the same folder the input file is. So you will have to get the parent directory of the input file and replace it with this name.